update_prevalence.py

```diff
@@ -911,99 +911,127 @@ def rolldown_vaccine_types(parent: Place, children: List[Place]):
 
+        # Keep a list of countries where an exception was raised so we can discard them rather than fail the whole update
+        failed_countries = list()
+
         for country in self.countries.values():
-            for state in country.states.values():
-                if state.test_positivity_rate is None and state.tests_in_past_week is not None:
-                    try:
-                        state.test_positivity_rate = state.cases_last_week / state.tests_in_past_week
-                    except ZeroDivisionError:
-                        print_and_log_to_sentry(
-                            f"Couldn't calculate {state.fullname}'s test positivity rate because there were no tests last week. {state}"
-                        )
-                        state.test_positivity_rate = None
-                if state.vaccines_by_type is not None:
-                    rolldown_vaccine_types(state, state.counties.values())
-                for county in state.counties.values():
-                    if county.population == 0 and county.name not in fake_names:
-                        raise ValueError(f"Missing population data for {county!r}")
-                rollup_population(state, "counties")
-                rollup_vaccines(state, "counties")
-            rollup_population(country, "states")
+            try:
+                for state in country.states.values():
+                    if state.test_positivity_rate is None and state.tests_in_past_week is not None:
+                        try:
+                            state.test_positivity_rate = state.cases_last_week / state.tests_in_past_week
+                        except ZeroDivisionError:
+                            print_and_log_to_sentry(
+                                f"Couldn't calculate {state.fullname}'s test positivity rate because there were no tests last week. {state}"
+                            )
+                            state.test_positivity_rate = None
+                    if state.vaccines_by_type is not None:
+                        rolldown_vaccine_types(state, state.counties.values())
+                    for county in state.counties.values():
+                        if county.population == 0 and county.name not in fake_names:
+                            raise ValueError(f"Missing population data for {county!r}")
+                    rollup_population(state, "counties")
+                    rollup_vaccines(state, "counties")
+                rollup_population(country, "states")
+            except ValueError as err:
+                failed_countries.append(dict(country_name=country.name, error=err))
+
+        # Delete any countries that had an error
+        for country_data in failed_countries:
+            print_and_log_to_sentry(
+                f"Discarding country {country_data['country_name']} due to error: {country_data['error']!r}"
+            )
+            del self.countries[country_data["country_name"]]
```

Comment:
I'm worried about the user experience if a real country starts having sync problems, would they be confused by suddenly not being able to find their Country? 

Could we do something that keeps the Country name but displays an error about not having data / puts them into manual entry mode / nudges them into manual entry mode?